rundeckapp/grails-app/controllers/rundeck/controllers/MenuController.groovy

```diff
@@ -2915,12 +2916,26 @@ Format is a string like `2d1h4n5s` using the following characters for time units
                     && extra.futureScheduledExecutions.size() > max) {
                 extra.futureScheduledExecutions = extra.futureScheduledExecutions[0..<max]
             }
+        } else if(request.api_version >= ApiVersions.V48) {
+            def map = scheduledExecutionService.nextOneTimeScheduledExecutions([scheduledExecution] as List<ScheduledExecution>)
+            extra.futureScheduledExecutions = map ? map.values().toList() : []
+        }
 
-            if(request.api_version >= ApiVersions.V48) {
-                def averageDuration = executionService.getAverageDuration(scheduledExecution.uuid)
-                if (averageDuration > 0) {
-                    extra.averageDuration = averageDuration
-                }
+        if(request.api_version >= ApiVersions.V48) {
+            if (scheduledExecution?.project) {
+                IRundeckProject rundeckProject =  frameworkService.getFrameworkProject(scheduledExecution.project)
+                Map properties = rundeckProject.getProjectProperties()
+
+                def isExecutionDisabledNow = properties[UpdateModeProjectService.CONF_PROJECT_DISABLE_EXECUTION] == 'true'
+                def isScheduleDisabledNow = properties[UpdateModeProjectService.CONF_PROJECT_DISABLE_SCHEDULE] == 'true'
```

Comment:
I think instead of importing the UpdateModeProjectService, you can reference `ScheduledExecutionService.CONF_PROJECT_DISABLE_EXECUTION` and `ScheduledExecutionService.CONF_PROJECT_DISABLE_SCHEDULE`